test/algorithms/test_is_valid.hpp

```diff
@@ -98,17 +97,18 @@ struct is_convertible_to_closed<Ring, bg::ring_tag, bg::open>
 template <typename Polygon>
 struct is_convertible_to_closed<Polygon, bg::polygon_tag, bg::open>
 {
-    typedef typename bg::ring_type<Polygon>::type ring_type;
+    using ring_type = typename bg::ring_type<Polygon>::type;
 
     template <typename InteriorRings>
     static inline
     bool apply_to_interior_rings(InteriorRings const& interior_rings)
     {
-        return bg::detail::check_iterator_range
-            <
-                is_convertible_to_closed<ring_type>
-            >::apply(boost::begin(interior_rings),
-                     boost::end(interior_rings));
+        return boost::end(interior_rings) == std::find_if(
+            boost::begin(interior_rings),
+            boost::end(interior_rings),
+            []( auto const& ring ){ return ! is_convertible_to_closed
+                <decltype(ring)>::apply(ring); }
```

Comment:
`decltype(ring)` is const reference is this what you wanted?

Also here different formatting of lambda expression is used.

Reply:
This is a mistake, should be `ring_type`